LibThreatClassic2.lua

```diff
@@ -795,6 +822,27 @@ function ThreatLib:ThreatUpdatedForUnit(unit_id, target_guid, threat)
 		self:UpdateTPS(unit_id, target_guid, threat)
 		self:_setThreat(unit_id, target_guid, threat)
 		_callbacks:Fire("ThreatUpdated", unit_id, target_guid, threat)
+
+		if unit_id == playerGUID then
+			lastMeleeRangeCheckTime = 0
+		end
+	end
+end
+
+-- #NODOC
+function ThreatLib:MeleeRangeUpdated(unitGUID, targetGUID, inMeleeRange)
+	self:MeleeRangeUpdatedForUnit(unitGUID, targetGUID, inMeleeRange)
+end
+
+-- #NODOC
+function ThreatLib:MeleeRangeUpdatedForUnit(unitGUID, targetGUID, inMeleeRange)
+	if type(unitGUID) ~= "string" then
+		error(("Assertion failed: type(%s --[[unitGUID]]) == %q, trace: %s"):format(self:toliteral(unitGUID), "string", debugstack()))
+	end
+
+	if self:_getInMeleeRange(unitGUID, targetGUID) ~= inMeleeRange then
+		self:_setInMeleeRange(unitGUID, targetGUID, inMeleeRange)
+		_callbacks:Fire("MeleeRangeUpdated", unitGUID, targetGUID, inMeleeRange)
```

Comment:
this callback is never used right?

Reply:
Melee range updates change the output of `UnitDetailedThreatSituation`. So anyone how want to be notified when this changes needs to register the `MeleeRangeUpdated` event.